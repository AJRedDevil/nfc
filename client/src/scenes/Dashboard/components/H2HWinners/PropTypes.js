import PropTypes from 'prop-types';

const data = PropTypes.shape({
  lastFetched: PropTypes.string,
  leagueNames: PropTypes.object,
  standings: PropTypes.object,
});

const head = PropTypes.arrayOf(PropTypes.shape({text: PropTypes.string}));

const winnersSchema = PropTypes.shape({
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  head,
});

const winners = PropTypes.arrayOf(PropTypes.array);

const h2hLeagueSchema = PropTypes.shape(head);

export default {data, h2hLeagueSchema, winners, winnersSchema};
