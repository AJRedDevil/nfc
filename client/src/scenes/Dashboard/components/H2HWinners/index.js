// npm packages
import React, {Component} from 'react';
import {connect} from 'react-redux';
import {func} from 'prop-types';
import {size} from 'lodash';

// our packages
import LoadingTable from '../../../../components/LoadingTable';
import TableHeading from '../../../../components/TableHeading';
import Table from '../../../../components/Table';
import {fetchH2HStandingsData} from '../../../../services/h2h/actions';
import H2HStandingsPropTypes from './PropTypes';
import Animation from '../../../../components/animations';

class H2HWinners extends Component {
  componentDidMount() {
    this.props.fetchH2HStandingsData();
  }

  renderH2HWinners = props => (
    <Animation.Fade in>
      <div>
        <TableHeading {...props.schema} />
        <Table body={props.winners} {...props.schema} />
      </div>
    </Animation.Fade>
  );

  render() {
    const {h2hStandingsData} = this.props;
    return size(h2hStandingsData.data) > 0 ? (
      this.renderH2HWinners(h2hStandingsData)
    ) : (
      <LoadingTable title="H2HWinners" />
    );
  }
}
H2HWinners.propTypes = {
  fetchH2HStandingsData: func.isRequired,
  h2hStandingsData: H2HStandingsPropTypes.data,
};

const mapStateToProps = state => ({
  h2hStandingsData: state.H2HStandings,
});

const mapDispatchToProps = dispatch => ({
  fetchH2HStandingsData: () => dispatch(fetchH2HStandingsData()),
});

export default connect(mapStateToProps, mapDispatchToProps)(H2HWinners);
