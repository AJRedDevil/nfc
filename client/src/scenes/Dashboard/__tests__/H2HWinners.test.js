// npm packages
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

// our packages
import H2HWinners from '../components/H2HWinners';
import DashboardMock from './mock.json';

// create mock store
const mockStore = configureMockStore();

test('# H2HWinners', () => {
  const store = mockStore({
    H2HStandings: DashboardMock.H2HStandings,
  });
  const wrapper = renderer.create(<H2HWinners store={store} />).toJSON();
  expect(wrapper).toMatchSnapshot();
});
