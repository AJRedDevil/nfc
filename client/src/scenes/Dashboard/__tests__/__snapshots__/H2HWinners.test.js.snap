// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# H2HWinners 1`] = `
<div>
  <section
    className="hero is-small is-info is-bold"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <h1
          className="title"
        >
          test
        </h1>
        <h2
          className="subtitle"
        />
      </div>
    </div>
  </section>
  <table
    className="table is-fullwidth is-striped is-narrow is-hoverable"
  >
    <thead>
      <tr>
        <th>
          Div
        </th>
        <th>
          Team
        </th>
        <th>
          Player
        </th>
        <th>
          Score
        </th>
      </tr>
    </thead>
    <tfoot>
      <tr />
    </tfoot>
    <tbody>
      <tr>
        <th>
          A
        </th>
        <td>
          Test_2
        </td>
        <td>
          Test 2
        </td>
        <td>
          100
        </td>
      </tr>
      <tr>
        <th>
          B
        </th>
        <td>
          Test_1
        </td>
        <td>
          Test 1
        </td>
        <td>
          70
        </td>
      </tr>
      <tr>
        <th>
          C
        </th>
        <td>
          Test_3
        </td>
        <td>
          Test 3
        </td>
        <td>
          60
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
